Defined Front Door redirect rule

WolfyUK opened the pull request:
### Context
[AB#260470](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/260470) [AB#260471](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/260471)

### Change proposed in this pull request
Created Front Door redirect rule to ensure all traffic to non-primary host names is sent to the root path of the primary host name.

### Guidance to review 
Deployed to `d11` feature environment to validate behaviour. Additional endpoint manually associated with Front Door profile in order to test non-primary host name redirection logic.

### Checklist (add/remove as appropriate)
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [ ] ~~You have run all unit/integration tests and they pass~~
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally
- [ ] ~~You have reviewed with UX/Design~~

J05h-L reviewed (approved):
lgtm

jrabbott commented:
Browsers can/do aggressively cache the permanent redirect. This will impact/impose future constraints

It isn't clear from the ticket why are doing this / what problem we are solving with this change?

WolfyUK commented:
> Browsers can/do aggressively cache the permanent redirect. This will impact/impose future constraints

Happy to switch to a `307 Temporary redirect`.

> It isn't clear from the ticket why are doing this / what problem we are solving with this change?

Discussions going on throughout yesterday and today have led to this requirement. It was also listed on the [original decision](https://github.com/DFE-Digital/education-benchmarking-and-insights/blob/main/documentation/architecture/decisions/0010-sfb-decomm.md#option-2-dns-level-redirect-to-the-new-service) but implementation was omitted as part of the decommissioning work intended to be deployed last release.

jrabbott commented:
I think there is some confusion. The document doesn't describe providing a redirect response from our front door.

Without full context (which is missing from the user story) 307 should be used to avoid caching issues.

WolfyUK commented:
> Without full context (which is missing from the user story) 307 should be used to avoid caching issues.

Switched to `307 Temporary redirect` and added more detail to the work item.

jrabbott commented:
Keep in mind the pervious implementation was intentional.

If the user navigated to an unsupported URL/route they would see a 404 error page and be able to take the necessary action.

With this implementation the user won't have this experience, and will be silently taken to the service home page without knowing why.

WolfyUK commented:
> With this implementation the user won't have this experience, and will be silently taken to the service home page without knowing why.

Spot on. This has been agreed with the client.

jrabbott commented:
Decision needs to be documented as this is a poor user experience.
